src/map/map.cpp

```diff
@@ -345,6 +345,31 @@ TIMER_FUNC(map_freeblock_timer){
 	return 0;
 }
 
+FreeBlockLock::FreeBlockLock(bool startLocked) {
+	if (startLocked) {
+		lock();
+	}
+}
+
+FreeBlockLock::~FreeBlockLock() {
+	unlock();
+}
+
+void FreeBlockLock::lock() {
+	if (!locked) {
+		map_freeblock_lock();
+		locked = true;
+	}
```

Comment:
Shouldnt we warn/error if it is called twice?

Reply:
Good idea, I'll make the change for `lock`